SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift

```diff
@@ -54,3 +54,43 @@ extension Array {
         }
     }
 }
+
+extension CollectionType {
+    public var ip_middleValues: [Generator.Element]? {
```

Comment:
My issue with this is that median does imply that it uses comparison, but we don't do any sorting -- we are really just grabbing the middle values. I made this separate because I thought grabbing the middle elements from an array would have some use outside of determining the median.